Index: org.talend.designer.components.localprovider/components/templates/db_output_bulk.skeleton
===================================================================
--- org.talend.designer.components.localprovider/components/templates/db_output_bulk.skeleton	(revision 29597)
+++ org.talend.designer.components.localprovider/components/templates/db_output_bulk.skeleton	(working copy)
@@ -139,7 +139,10 @@
             this.sqlStmt = sql;
         }
         public String getSqlStmt() {
-            return this.sqlStmt;
+            if (this.getColumn().getTalendType().equals("id_Geometry"))
+               return "GeomFromText(?, ?)";    // For PostGIS
+            else
+				return this.sqlStmt;
         }
         public void setValue(String value) {
             this.value = value;
@@ -715,9 +718,13 @@
                 } else {
                     setStmt.append(prefix + cid + ".setNull(" + index + ", java.sql.Types.VARCHAR);\r\n");                    
                 }                
-            } else if(typeToGenerate.equals("Object")) {
+            } else if(typeToGenerate.equals("Geometry")) {
                 isObject = true;
                 setStmt.append("if(" + incomingConnName + "." + column.getName() + " == null) {\r\n");
+                setStmt.append(prefix + cid + ".setNull(" + index + ", java.sql.Types.NULL);\r\n");
+			} else if(typeToGenerate.equals("Object")) {
+                isObject = true;
+                setStmt.append("if(" + incomingConnName + "." + column.getName() + " == null) {\r\n");
                 if(dbType != null && dbType.equals("BLOB")) {
                     setStmt.append(prefix + cid + ".setNull(" + index + ", java.sql.Types.BLOB);\r\n");
                 } else {
@@ -843,7 +850,26 @@
                 setStmt.append("} else {\r\n");
                 setStmt.append(prefix + cid + ".setNull(" + index + ", java.sql.Types.DATE);\r\n");
                 setStmt.append("}\r\n");
-            } else {
+            } else if(typeToGenerate.equals("Geometry")) {
+                // Load geometry
+                setStmt.append(
+                                prefix +
+                                cid +
+                                ".setString(" + 
+                                    index +
+                                    ", ((org.talend.sdi.geometry.Geometry)(" +
+                                    incomingConnName + "." + column.getName() + ")).toString()" +
+                                    ");\r\n");
+                // TODO : should use forceCoordinate first, then getSRID method or -1 if it's null.
+                setStmt.append(
+                                prefix +
+                                cid +
+                                ".setInt(" + 
+                                    index + 
+                                    " + 1, ((org.talend.sdi.geometry.Geometry)(" + 
+                                    incomingConnName + "." + column.getName() + ")).getSRID()" +
+                                    ");\r\n");
+			} else {
                 setStmt.append(prefix + cid + ".set" + typeToGenerate + "(" + index + ", " + incomingConnName + "." + column.getName() + ");\r\n");
             }
             if(isObject) {
Index: org.talend.designer.components.localprovider/build.properties
===================================================================
--- org.talend.designer.components.localprovider/build.properties	(revision 29597)
+++ org.talend.designer.components.localprovider/build.properties	(working copy)
@@ -7,3 +7,6 @@
                plugin.properties
 javacSource=1.5
 javacTarget=1.5
+bin.excludes = components/ext/sdi/,\
+               components/ext/sdisandbox/,\
+               components/ext/sdisextante/
