Index: org.talend.designer.codegen/resources/subprocess_header.javajet
===================================================================
--- org.talend.designer.codegen/resources/subprocess_header.javajet	(revision 37513)
+++ org.talend.designer.codegen/resources/subprocess_header.javajet	(working copy)
@@ -327,7 +327,7 @@
                				this.<%=column.getLabel() %> = byteArray;
            				}
 					<%
-				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
+				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("Geometry") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
 			 		hasAtLeastOneObjectType = true;
 					%>
 						this.<%=column.getLabel() %> = (<%= typeToGenerate %>) dis.readObject();
@@ -428,7 +428,7 @@
                				dos.write(this.<%=column.getLabel() %>);
 		            	}
 					<%
-				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
+				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("Geometry") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
 					%>
        			    	dos.writeObject(this.<%=column.getLabel() %>);
 					<%
@@ -525,7 +525,7 @@
            			    	this.<%=column.getLabel() %> = new Date(dis.readLong());
            				}
 					<%
-				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
+				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("Geometry") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
 			 		hasAtLeastOneObjectType = true;
 					%>
        			    	this.<%=column.getLabel() %> = (<%= typeToGenerate %>) ois.readObject();
@@ -621,7 +621,7 @@
            			    	dos.writeLong(this.<%=column.getLabel() %>.getTime());
 		            	}
 					<%
-				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
+				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("Geometry") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
 					%>
        			    	oos.writeObject(this.<%=column.getLabel() %>);
 					<%
