Index: org.talend.libraries/build.properties
===================================================================
--- org.talend.libraries/build.properties	(revision 29597)
+++ org.talend.libraries/build.properties	(working copy)
@@ -102,7 +102,8 @@
                commons-httpclient-3.1.jar,\
                jackson-0.9.2.jar,\
                jxplorer.jar,\
-               salesforceCRMManagement.jar
+               salesforceCRMManagement.jar,\
+               org.talend.sdi.jar
 source.. = src/main/java/
 src.includes = src/main/java/,\
                jakarta-oro-2.0.8.jar
Index: org.talend.libraries/.classpath
===================================================================
--- org.talend.libraries/.classpath	(revision 29597)
+++ org.talend.libraries/.classpath	(working copy)
@@ -5,6 +5,7 @@
 	<classpathentry exported="true" kind="lib" path="jackson-0.9.2.jar"/>
 	<classpathentry exported="true" kind="lib" path="commons-httpclient-3.1.jar"/>
 	<classpathentry exported="true" kind="lib" path="fakejdbc_driver.jar"/>
+	<classpathentry kind="lib" path="org.talend.sdi.jar"/>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
 	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
 	<classpathentry kind="src" path="src/main/java"/>
Index: org.talend.libraries/META-INF/MANIFEST.MF
===================================================================
--- org.talend.libraries/META-INF/MANIFEST.MF	(revision 29597)
+++ org.talend.libraries/META-INF/MANIFEST.MF	(working copy)
@@ -46,7 +46,8 @@
  commons-httpclient-3.1.jar,
  jackson-0.9.2.jar,
  jxplorer.jar,
- salesforceCRMManagement.jar
+ salesforceCRMManagement.jar,
+ org.talend.sdi.jar
 Eclipse-LazyStart: true
 Require-Bundle: org.apache.log4j,
  org.eclipse.core.runtime,
@@ -64,7 +65,8 @@
  javax.xml.soap,
  org.talend.libraries.jdbc.ingres;visibility:=reexport,
  org.talend.libraries.jdbc.hsql;visibility:=reexport,
- org.talend.libraries.jdbc.informix;visibility:=reexport
+ org.talend.libraries.jdbc.informix;visibility:=reexport,
+ org.talend.libraries.sdi;bundle-version="0.0.1"
 Bundle-Activator: org.talend.libraries.LibrariesPlugin
 Import-Package: com.ibm.icu.text
 Bundle-Vendor: .Talend SA.
@@ -438,6 +440,7 @@
  org.talend.fileprocess,
  org.talend.fileprocess.delimited,
  org.talend.libraries,
+ org.talend.sdi.geometry,
  org.w3c.dom,
  org.w3c.dom.events,
  org.w3c.dom.html,
