Index: META-INF/MANIFEST.MF
===================================================================
--- META-INF/MANIFEST.MF	(revision 26217)
+++ META-INF/MANIFEST.MF	(working copy)
@@ -4,7 +4,8 @@
 Bundle-SymbolicName: org.talend.designer.codegen;singleton:=true
 Bundle-Version: 2.2.0.qualifier
 Bundle-Localization: plugin
-Require-Bundle: org.talend.libraries.logging, org.eclipse.core.resources,
+Require-Bundle: org.talend.libraries.logging,
+ org.eclipse.core.resources,
  org.eclipse.core.runtime,
  org.eclipse.emf.codegen,
  org.eclipse.emf.common,
@@ -23,3 +24,5 @@
  org.talend.designer.codegen.exception
 Bundle-Vendor: .Talend SA.
 Bundle-Activator: org.talend.designer.codegen.CodeGeneratorActivator
+Bundle-ClassPath: org.talend.sdi.jar,
+ .
Index: resources/subprocess_header.javajet
===================================================================
--- resources/subprocess_header.javajet	(revision 26217)
+++ resources/subprocess_header.javajet	(working copy)
@@ -289,7 +289,7 @@
                				this.<%=column.getLabel() %> = byteArray;
            				}
 					<%
-				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
+				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("Geometry") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
 			 		hasAtLeastOneObjectType = true;
 					%>
 						this.<%=column.getLabel() %> = (<%= typeToGenerate %>) dis.readObject();
@@ -390,7 +390,7 @@
                				dos.write(this.<%=column.getLabel() %>);
 		            	}
 					<%
-				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
+				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("Geometry") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
 					%>
        			    	dos.writeObject(this.<%=column.getLabel() %>);
 					<%
@@ -487,7 +487,7 @@
            			    	this.<%=column.getLabel() %> = new Date(dis.readLong());
            				}
 					<%
-				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
+				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("Geometry") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
 			 		hasAtLeastOneObjectType = true;
 					%>
        			    	this.<%=column.getLabel() %> = (<%= typeToGenerate %>) ois.readObject();
@@ -583,7 +583,7 @@
            			    	dos.writeLong(this.<%=column.getLabel() %>.getTime());
 		            	}
 					<%
-				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
+				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("Geometry") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
 					%>
        			    	oos.writeObject(this.<%=column.getLabel() %>);
 					<%
Index: resources/header.javajet
===================================================================
--- resources/header.javajet	(revision 26217)
+++ resources/header.javajet	(working copy)
@@ -74,6 +74,8 @@
 import java.io.IOException;
 import java.util.Comparator;
 
+import org.talend.sdi.geometry.Geometry;
+
 <%
 	List<INode> nodesWithImport = process.getNodesWithImport();
 	if(nodesWithImport != null) {
@@ -382,10 +384,13 @@
             <%
 		}
 	}
+	
+	List<String> beginNodes = new ArrayList<String>();
+	
 	for (INode node : processNodes) {
-		if (node.isDesignSubjobStartNode()) {
+		if (!beginNodes.contains(node.getDesignSubjobStartNode().getUniqueName())) {
 %>
-			public void <%=node.getUniqueName() %>_onSubJobError(Exception exception, final java.util.Map<String, Object> globalMap) throws TalendException {
+			public void <%=node.getDesignSubjobStartNode().getUniqueName() %>_onSubJobError(Exception exception, final java.util.Map<String, Object> globalMap) throws TalendException {
 <%
 			List< ? extends IConnection> conns = node.getOutgoingConnections();
 			for (IConnection conn : conns) {
@@ -413,6 +418,7 @@
 <%
 				}
 			}
+			beginNodes.add(node.getDesignSubjobStartNode().getUniqueName());
 %>
 			}
 <%
Index: build.properties
===================================================================
--- build.properties	(revision 26217)
+++ build.properties	(working copy)
@@ -4,6 +4,8 @@
                META-INF/,\
                .,\
                icons/,\
-               resources/
+               resources/,\
+               org.talend.sdi.jar
 javacSource=1.5
 javacTarget=1.5
+source.org.talend.sdi.jar = src/main/java/
\ No newline at end of file
