Index: org.talend.designer.codegen/META-INF/MANIFEST.MF
===================================================================
--- org.talend.designer.codegen/META-INF/MANIFEST.MF	(revision 29597)
+++ org.talend.designer.codegen/META-INF/MANIFEST.MF	(working copy)
@@ -25,3 +25,4 @@
  org.talend.designer.codegen.exception
 Bundle-Vendor: .Talend SA.
 Bundle-Activator: org.talend.designer.codegen.CodeGeneratorActivator
+Bundle-ClassPath: org.talend.sdi.jar, .
Index: org.talend.designer.codegen/resources/subprocess_header.javajet
===================================================================
--- org.talend.designer.codegen/resources/subprocess_header.javajet	(revision 29597)
+++ org.talend.designer.codegen/resources/subprocess_header.javajet	(working copy)
@@ -317,7 +317,7 @@
                				this.<%=column.getLabel() %> = byteArray;
            				}
 					<%
-				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
+				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("Geometry") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
 			 		hasAtLeastOneObjectType = true;
 					%>
 						this.<%=column.getLabel() %> = (<%= typeToGenerate %>) dis.readObject();
@@ -418,7 +418,7 @@
                				dos.write(this.<%=column.getLabel() %>);
 		            	}
 					<%
-				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
+				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("Geometry") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
 					%>
        			    	dos.writeObject(this.<%=column.getLabel() %>);
 					<%
@@ -515,7 +515,7 @@
            			    	this.<%=column.getLabel() %> = new Date(dis.readLong());
            				}
 					<%
-				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
+				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("Geometry") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
 			 		hasAtLeastOneObjectType = true;
 					%>
        			    	this.<%=column.getLabel() %> = (<%= typeToGenerate %>) ois.readObject();
@@ -611,7 +611,7 @@
            			    	dos.writeLong(this.<%=column.getLabel() %>.getTime());
 		            	}
 					<%
-				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
+				} else if(typeToGenerate.equals("Object") || typeToGenerate.equals("Geometry") || typeToGenerate.equals("BigDecimal") || typeToGenerate.equals("List")) {
 					%>
        			    	oos.writeObject(this.<%=column.getLabel() %>);
 					<%
Index: org.talend.designer.codegen/resources/header.javajet
===================================================================
--- org.talend.designer.codegen/resources/header.javajet	(revision 29597)
+++ org.talend.designer.codegen/resources/header.javajet	(working copy)
@@ -73,6 +73,7 @@
 import java.io.ObjectInputStream;
 import java.io.IOException;
 import java.util.Comparator;
+import org.talend.sdi.geometry.Geometry;
 
 <%
 	List<INode> nodesWithImport = process.getNodesWithImport();
@@ -396,10 +397,13 @@
             <%
 		}
 	}
+	
+	List<String> beginNodes = new ArrayList<String>();
+	
 	for (INode node : processNodes) {
-		if (node.isDesignSubjobStartNode()) {
+		if (!beginNodes.contains(node.getDesignSubjobStartNode().getUniqueName())) {
 %>
-			public void <%=node.getUniqueName() %>_onSubJobError(Exception exception, final java.util.Map<String, Object> globalMap) throws TalendException {
+			public void <%=node.getDesignSubjobStartNode().getUniqueName() %>_onSubJobError(Exception exception, final java.util.Map<String, Object> globalMap) throws TalendException {
 <%
 			List< ? extends IConnection> conns = node.getOutgoingConnections();
 			for (IConnection conn : conns) {
@@ -430,6 +434,7 @@
 %>
 			}
 <%
+			beginNodes.add(node.getDesignSubjobStartNode().getUniqueName());
 		}
 	}
-%>
+%>
\ No newline at end of file
Index: org.talend.designer.codegen/build.properties
===================================================================
--- org.talend.designer.codegen/build.properties	(revision 29597)
+++ org.talend.designer.codegen/build.properties	(working copy)
@@ -5,6 +5,8 @@
                .,\
                icons/,\
                resources/,\
-               jet_stub/
+               jet_stub/,\
+               org.talend.sdi.jar
 javacSource=1.5
 javacTarget=1.5
+source.org.talend.sdi.jar = src/main/java/
\ No newline at end of file
