Index: org.talend.repository/src/main/java/messages.properties
===================================================================
--- org.talend.repository/src/main/java/messages.properties	(revision 29597)
+++ org.talend.repository/src/main/java/messages.properties	(working copy)
@@ -713,6 +713,8 @@
 NewProjectWizardPage.technicalName=Technical Name
 NewProjectWizardPage.title=Project
 NewRoutineWizard.title=New routine
+NewProjectWizardPage.JavaMsg=
+NewProjectWizardPage.perlMsg=(No geospatial components available)
 NewRoutineWizardPage.description=Add a new routine to repository
 NewRoutineWizardPage.title=New routine
 NewSqlpatternWizard.failureTitle=Error
Index: org.talend.repository/src/main/java/org/talend/repository/ui/wizards/newproject/NewProjectWizardPage.java
===================================================================
--- org.talend.repository/src/main/java/org/talend/repository/ui/wizards/newproject/NewProjectWizardPage.java	(revision 29597)
+++ org.talend.repository/src/main/java/org/talend/repository/ui/wizards/newproject/NewProjectWizardPage.java	(working copy)
@@ -140,12 +140,13 @@
         gridLayout.marginHeight = 0;
         radioContainer.setLayout(gridLayout);
 
+        languageJavaRadio = new Button(radioContainer, SWT.RADIO);
+        languageJavaRadio.setText(ECodeLanguage.JAVA.getName() + Messages.getString("NewProjectWizardPage.JavaMsg"));
+        
         languagePerlRadio = new Button(radioContainer, SWT.RADIO);
-        languagePerlRadio.setText(ECodeLanguage.PERL.getName());
-
-        languageJavaRadio = new Button(radioContainer, SWT.RADIO);
-        languageJavaRadio.setText(ECodeLanguage.JAVA.getName());
-
+        languagePerlRadio.setText(ECodeLanguage.PERL.getName() + Messages.getString("NewProjectWizardPage.perlMsg"));
+         
+        
         IBrandingService brandingService = (IBrandingService) GlobalServiceRegister.getDefault().getService(
                 IBrandingService.class);
         String[] availableLanguages = brandingService.getBrandingConfiguration().getAvailableLanguages();
