Index: org.eclipse.datatools.connectivity/META-INF/MANIFEST.MF
===================================================================
--- org.eclipse.datatools.connectivity/META-INF/MANIFEST.MF	(revision 31214)
+++ org.eclipse.datatools.connectivity/META-INF/MANIFEST.MF	(working copy)
@@ -8,7 +8,7 @@
 Bundle-Localization: plugin
 Require-Bundle: org.eclipse.core.runtime;bundle-version="[3.2.0,4.0.0)",
  org.eclipse.core.resources;bundle-version="[3.2.0,4.0.0)",
- com.ibm.icu;bundle-version="[3.4.0,4.0.0)"
+ com.ibm.icu;bundle-version="[3.4.0,4.0.2)"
 Eclipse-LazyStart: true
 Export-Package: org.eclipse.datatools.connectivity,
  org.eclipse.datatools.connectivity.drivers,
