Index: org.talend.libraries/build.properties
===================================================================
--- org.talend.libraries/build.properties	(revision 31214)
+++ org.talend.libraries/build.properties	(working copy)
@@ -107,7 +107,8 @@
                ooxml-schemas-1.0.jar,\
                poi-3.5-beta5-20090219.jar,\
                poi-ooxml-3.5-beta5-20090219.jar,\
-               xmlbeans-2.3.0.jar
+               xmlbeans-2.3.0.jar,\
+               org.talend.sdi.jar
 source.. = src/main/java/
 src.includes = src/main/java/,\
                jakarta-oro-2.0.8.jar
Index: org.talend.libraries/.classpath
===================================================================
--- org.talend.libraries/.classpath	(revision 31214)
+++ org.talend.libraries/.classpath	(working copy)
@@ -5,6 +5,7 @@
 	<classpathentry kind="src" path="src/main/java"/>
 	<classpathentry exported="true" kind="lib" path="junit-4.0.jar" sourcepath="org.talend.librariessrc.zip"/>
 	<classpathentry exported="true" kind="lib" path="dom4j-1.6.1.jar" sourcepath="org.talend.librariessrc.zip"/>
+	<classpathentry exported="true" kind="lib" path="nlog4j-1.2.25.jar"/>
 	<classpathentry exported="true" kind="lib" path="castor-1.0.3.jar" sourcepath="libs_src/castor-1.0.3"/>
 	<classpathentry exported="true" kind="lib" path="xerces-J_1.4.0.jar" sourcepath="org.talend.librariessrc.zip"/>
 	<classpathentry exported="true" kind="lib" path="nanoxml.jar" sourcepath="org.talend.librariessrc.zip"/>
@@ -28,7 +29,6 @@
 	<classpathentry exported="true" kind="lib" path="org.apache.directory.studio.jars_1.1.0.v20080331.jar" sourcepath="org.talend.librariessrc.zip"/>
 	<classpathentry exported="true" kind="lib" path="shared-ldap-0.9.8.jar" sourcepath="org.talend.librariessrc.zip"/>
 	<classpathentry exported="true" kind="lib" path="shared-ldap-constants-0.9.8.jar" sourcepath="org.talend.librariessrc.zip"/>
-	<classpathentry exported="true" kind="lib" path="nlog4j-1.2.25.jar" sourcepath="org.talend.librariessrc.zip"/>
 	<classpathentry exported="true" kind="lib" path="jaxen-1.1.1.jar" sourcepath="org.talend.librariessrc.zip"/>
 	<classpathentry exported="true" kind="lib" path="wsdl4j-1.5.1.jar" sourcepath="org.talend.librariessrc.zip"/>
 	<classpathentry exported="true" kind="lib" path="paraccel-jdbc.jar" sourcepath="org.talend.librariessrc.zip"/>
@@ -43,5 +43,6 @@
 	<classpathentry exported="true" kind="lib" path="poi-3.5-beta5-20090219.jar" sourcepath="org.talend.librariessrc.zip"/>
 	<classpathentry exported="true" kind="lib" path="poi-ooxml-3.5-beta5-20090219.jar" sourcepath="org.talend.librariessrc.zip"/>
 	<classpathentry exported="true" kind="lib" path="xmlbeans-2.3.0.jar" sourcepath="org.talend.librariessrc.zip"/>
+	<classpathentry kind="lib" path="org.talend.sdi.jar"/>
 	<classpathentry kind="output" path="class"/>
 </classpath>
Index: org.talend.libraries/META-INF/MANIFEST.MF
===================================================================
--- org.talend.libraries/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.libraries/META-INF/MANIFEST.MF	(working copy)
@@ -51,7 +51,8 @@
  ooxml-schemas-1.0.jar,
  poi-3.5-beta5-20090219.jar,
  poi-ooxml-3.5-beta5-20090219.jar,
- xmlbeans-2.3.0.jar
+ xmlbeans-2.3.0.jar,
+ org.talend.sdi.jar
 Eclipse-LazyStart: true
 Require-Bundle: org.apache.log4j;bundle-version="1.2.13",
  org.eclipse.core.runtime,
@@ -69,7 +70,8 @@
  javax.xml.soap,
  org.talend.libraries.jdbc.ingres;visibility:=reexport,
  org.talend.libraries.jdbc.hsql;visibility:=reexport,
- org.talend.libraries.jdbc.informix;visibility:=reexport
+ org.talend.libraries.jdbc.informix;visibility:=reexport,
+ org.talend.libraries.sdi
 Bundle-Activator: org.talend.libraries.LibrariesPlugin
 Import-Package: com.ibm.icu.text
 Bundle-Vendor: .Talend SA.
@@ -297,6 +299,7 @@
  org.apache.directory.studio.ldifparser.model.lines,
  org.apache.directory.studio.ldifparser.parser,
  org.apache.html.dom,
+ org.apache.log4j,
  org.apache.poi,
  org.apache.poi.common.usermodel,
  org.apache.poi.ddf,
@@ -542,6 +545,7 @@
  org.talend.fileprocess,
  org.talend.fileprocess.delimited,
  org.talend.libraries,
+ org.talend.sdi.geometry,
  org.w3c.dom,
  org.w3c.dom.events,
  org.w3c.dom.html,
@@ -593,3 +597,4 @@
  schemaorg_apache_xmlbeans.system.sXMLLANG,
  schemaorg_apache_xmlbeans.system.sXMLSCHEMA,
  schemaorg_apache_xmlbeans.system.sXMLTOOLS
+Eclipse-BundleShape: dir
Index: org.talend.libraries.apache/META-INF/MANIFEST.MF
===================================================================
--- org.talend.libraries.apache/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.libraries.apache/META-INF/MANIFEST.MF	(working copy)
@@ -83,3 +83,4 @@
 Bundle-Vendor: .Talend SA.
 Eclipse-BuddyPolicy: dependent
 Require-Bundle: org.apache.log4j;bundle-version="1.2.13";resolution:=optional
+Eclipse-BundleShape: dir
Index: org.talend.libraries.hibernate/META-INF/MANIFEST.MF
===================================================================
--- org.talend.libraries.hibernate/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.libraries.hibernate/META-INF/MANIFEST.MF	(working copy)
@@ -198,3 +198,4 @@
  org.xml.sax.ext,
  org.xml.sax.helpers
 Eclipse-BuddyPolicy: dependent
+Eclipse-BundleShape: dir
Index: org.talend.libraries.javacsv/META-INF/MANIFEST.MF
===================================================================
--- org.talend.libraries.javacsv/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.libraries.javacsv/META-INF/MANIFEST.MF	(working copy)
@@ -8,3 +8,4 @@
 Bundle-ClassPath: lib/javacsv.jar,
  .
 Export-Package: com.csvreader
+Eclipse-BundleShape: dir
Index: org.talend.libraries.jdbc.as400/META-INF/MANIFEST.MF
===================================================================
--- org.talend.libraries.jdbc.as400/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.libraries.jdbc.as400/META-INF/MANIFEST.MF	(working copy)
@@ -18,3 +18,4 @@
 Bundle-ClassPath: .,
  lib/jt400_V5R3.jar
 Bundle-Vendor: .Talend SA.
+Eclipse-BundleShape: dir
Index: org.talend.libraries.jdbc.db2/META-INF/MANIFEST.MF
===================================================================
--- org.talend.libraries.jdbc.db2/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.libraries.jdbc.db2/META-INF/MANIFEST.MF	(working copy)
@@ -30,3 +30,5 @@
  sqlj.runtime.profile.util,
  sqlj.runtime.ref,
  sqlj.runtime.util
+Eclipse-BundleShape: dir
+
Index: org.talend.libraries.jdbc.hsql/META-INF/MANIFEST.MF
===================================================================
--- org.talend.libraries.jdbc.hsql/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.libraries.jdbc.hsql/META-INF/MANIFEST.MF	(working copy)
@@ -19,3 +19,4 @@
  org.hsqldb.types,
  org.hsqldb.util
 Bundle-RequiredExecutionEnvironment: J2SE-1.5
+Eclipse-BundleShape: dir
Index: org.talend.libraries.jdbc.informix/META-INF/MANIFEST.MF
===================================================================
--- org.talend.libraries.jdbc.informix/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.libraries.jdbc.informix/META-INF/MANIFEST.MF	(working copy)
@@ -45,3 +45,4 @@
  udtudrmgr,
  UtilXML
 Bundle-Vendor: .Talend SA.
+Eclipse-BundleShape: dir
Index: org.talend.libraries.jdbc.ingres/META-INF/MANIFEST.MF
===================================================================
--- org.talend.libraries.jdbc.ingres/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.libraries.jdbc.ingres/META-INF/MANIFEST.MF	(working copy)
@@ -12,3 +12,4 @@
  com.ingres.gcf.util,
  com.ingres.jdbc
 Bundle-Vendor: .Talend SA.
+Eclipse-BundleShape: dir
Index: org.talend.libraries.jdbc.jtds/META-INF/MANIFEST.MF
===================================================================
--- org.talend.libraries.jdbc.jtds/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.libraries.jdbc.jtds/META-INF/MANIFEST.MF	(working copy)
@@ -12,3 +12,4 @@
  net.sourceforge.jtds.jdbcx.proxy,
  net.sourceforge.jtds.ssl,
  net.sourceforge.jtds.util
+Eclipse-BundleShape: dir
Index: org.talend.libraries.jdbc.mysql/META-INF/MANIFEST.MF
===================================================================
--- org.talend.libraries.jdbc.mysql/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.libraries.jdbc.mysql/META-INF/MANIFEST.MF	(working copy)
@@ -17,3 +17,4 @@
  com.mysql.jdbc.profiler,
  com.mysql.jdbc.util,
  org.gjt.mm.mysql
+Eclipse-BundleShape: dir
Index: org.talend.libraries.jdbc.oracle/META-INF/MANIFEST.MF
===================================================================
--- org.talend.libraries.jdbc.oracle/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.libraries.jdbc.oracle/META-INF/MANIFEST.MF	(working copy)
@@ -33,3 +33,4 @@
  oracle.sql,
  oracle.sql.converter,
  oracle.sql.converter_xcharset
+Eclipse-BundleShape: dir
Index: org.talend.libraries.jdbc.postgresql/META-INF/MANIFEST.MF
===================================================================
--- org.talend.libraries.jdbc.postgresql/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.libraries.jdbc.postgresql/META-INF/MANIFEST.MF	(working copy)
@@ -23,3 +23,4 @@
  org.postgresql.translation,
  org.postgresql.util,
  org.postgresql.xa
+Eclipse-BundleShape: dir
Index: org.talend.libraries.jdbc.sqlite3/META-INF/MANIFEST.MF
===================================================================
--- org.talend.libraries.jdbc.sqlite3/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.libraries.jdbc.sqlite3/META-INF/MANIFEST.MF	(working copy)
@@ -10,3 +10,4 @@
  org.ibex.nestedvm.util,
  org.sqlite
 Bundle-Vendor: .Talend SA.
+Eclipse-BundleShape: dir
Index: org.talend.libraries.jdbc.sybase/META-INF/MANIFEST.MF
===================================================================
--- org.talend.libraries.jdbc.sybase/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.libraries.jdbc.sybase/META-INF/MANIFEST.MF	(working copy)
@@ -13,3 +13,4 @@
  com.sybase.jdbc3.utils,
  com.sybase.jdbc3.utils.resource,
  com.sybase.jdbcx
+Eclipse-BundleShape: dir
Index: org.talend.libraries.jdbc.teradata/META-INF/MANIFEST.MF
===================================================================
--- org.talend.libraries.jdbc.teradata/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.libraries.jdbc.teradata/META-INF/MANIFEST.MF	(working copy)
@@ -34,3 +34,4 @@
  com.ncr.teradata.jtdgss,
  com.ncr.teradata.resource,
  org.ietf.jgssspi
+Eclipse-BundleShape: dir
Index: org.talend.libraries.jfreechart/META-INF/MANIFEST.MF
===================================================================
--- org.talend.libraries.jfreechart/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.libraries.jfreechart/META-INF/MANIFEST.MF	(working copy)
@@ -72,3 +72,4 @@
  org.jfree.ui.tabbedui,
  org.jfree.util
 Require-Bundle: org.eclipse.swt;bundle-version="3.4.0";resolution:=optional
+Eclipse-BundleShape: dir
Index: org.talend.librariesmanager/META-INF/MANIFEST.MF
===================================================================
--- org.talend.librariesmanager/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.librariesmanager/META-INF/MANIFEST.MF	(working copy)
@@ -14,9 +14,11 @@
  org.talend.libraries,
  org.eclipse.core.resources,
  org.eclipse.jdt.core,
- org.talend.commons.ui
+ org.talend.commons.ui,
+ org.talend.libraries.sdi
 Eclipse-LazyStart: true
 Export-Package: org.talend.librariesmanager.model,
  org.talend.librariesmanager.prefs,
  routines,
  routines.system
+Bundle-ClassPath: org.talend.sdi.jar
Index: org.talend.librariesmanager/build.properties
===================================================================
--- org.talend.librariesmanager/build.properties	(revision 31214)
+++ org.talend.librariesmanager/build.properties	(working copy)
@@ -1,9 +1,9 @@
-source.. = src/main/java/,\
-           resources/java/
 output.. = bin/
 bin.includes = META-INF/,\
-               .,\
                plugin.xml,\
                resources/,\
                icons/,\
-               plugin.properties
+               plugin.properties,\
+               org.talend.sdi.jar
+source.org.talend.sdi.jar = src/main/java/,\
+                            resources/java/
\ No newline at end of file
