Index: org.talend.repository/META-INF/MANIFEST.MF
===================================================================
--- org.talend.repository/META-INF/MANIFEST.MF	(revision 31214)
+++ org.talend.repository/META-INF/MANIFEST.MF	(working copy)
@@ -6,7 +6,8 @@
 Bundle-Localization: plugin
 Eclipse-LazyStart: true
 Bundle-Vendor: .Talend SA.
-Require-Bundle: org.talend.libraries,
+Require-Bundle: org.apache.log4j;bundle-version="1.2.13",
+ org.talend.libraries,
  org.apache.axis,
  org.talend.commons,
  org.talend.core,
@@ -31,8 +32,7 @@
  org.eclipse.ui.browser,
  org.talend.commons.ui,
  org.eclipse.core.filesystem,
- org.talend.registration;visibility:=reexport,
- org.apache.log4j;bundle-version="1.2.13"
+ org.talend.registration;visibility:=reexport
 Export-Package: org.talend.repository,
  org.talend.repository.constants,
  org.talend.repository.i18n,
