Index: org.eclipse.datatools.enablement.oda.xml/META-INF/MANIFEST.MF
===================================================================
--- org.eclipse.datatools.enablement.oda.xml/META-INF/MANIFEST.MF	(revision 31214)
+++ org.eclipse.datatools.enablement.oda.xml/META-INF/MANIFEST.MF	(working copy)
@@ -7,7 +7,7 @@
 Bundle-Vendor: Eclipse.org
 Bundle-Localization: plugin
 Require-Bundle: org.eclipse.datatools.connectivity.oda;bundle-version="[3.0.1,4.0.0)",
- com.ibm.icu;bundle-version="[3.4.4,4.0.0)",
+ com.ibm.icu;bundle-version="[3.4.4,4.0.2)",
  org.eclipse.swt,
  org.eclipse.jface,
  org.apache.xerces
