Index: org.talend.core/src/main/java/org/talend/core/model/metadata/types/JavaTypesManager.java
===================================================================
--- org.talend.core/src/main/java/org/talend/core/model/metadata/types/JavaTypesManager.java	(revision 29597)
+++ org.talend.core/src/main/java/org/talend/core/model/metadata/types/JavaTypesManager.java	(working copy)
@@ -76,6 +76,8 @@
 
     public static final JavaType OBJECT = new JavaType(Object.class);
 
+    public static final JavaType GEOMETRY = new JavaType(org.talend.sdi.geometry.Geometry.class);
+    
     public static final JavaType LIST = new JavaType(List.class);
 
     public static final JavaType BIGDECIMAL = new JavaType(BigDecimal.class);
@@ -83,7 +85,7 @@
     public static final JavaType PASSWORD = new JavaType(String.class, true, "Password"); //$NON-NLS-1$
 
     public static final JavaType[] JAVA_TYPES = new JavaType[] { BOOLEAN, BYTE, BYTE_ARRAY, CHARACTER, DATE, DOUBLE, FLOAT,
-            BIGDECIMAL, INTEGER, LONG, OBJECT, SHORT, STRING, LIST };
+            BIGDECIMAL, INTEGER, LONG, OBJECT, GEOMETRY, SHORT, STRING, LIST };
 
     private static Map<String, JavaType> shortNameToJavaType;
 
Index: org.talend.core/src/main/java/org/talend/core/model/metadata/types/ContextParameterJavaTypeManager.java
===================================================================
--- org.talend.core/src/main/java/org/talend/core/model/metadata/types/ContextParameterJavaTypeManager.java	(revision 29597)
+++ org.talend.core/src/main/java/org/talend/core/model/metadata/types/ContextParameterJavaTypeManager.java	(working copy)
@@ -31,7 +31,7 @@
 
     public static final JavaType[] JAVA_TYPES = new JavaType[] { JavaTypesManager.BOOLEAN, JavaTypesManager.CHARACTER,
             JavaTypesManager.DATE, JavaTypesManager.DOUBLE, JavaTypesManager.FLOAT, JavaTypesManager.INTEGER,
-            JavaTypesManager.LONG, JavaTypesManager.SHORT, JavaTypesManager.STRING, JavaTypesManager.OBJECT,
+            JavaTypesManager.LONG, JavaTypesManager.SHORT, JavaTypesManager.STRING, JavaTypesManager.OBJECT, JavaTypesManager.GEOMETRY,
             JavaTypesManager.BIGDECIMAL, JavaTypesManager.FILE, JavaTypesManager.DIRECTORY, JavaTypesManager.VALUE_LIST,
             JavaTypesManager.PASSWORD };
 
Index: org.talend.core/META-INF/MANIFEST.MF
===================================================================
--- org.talend.core/META-INF/MANIFEST.MF	(revision 29597)
+++ org.talend.core/META-INF/MANIFEST.MF	(working copy)
@@ -27,7 +27,8 @@
  org.eclipse.ui.views.properties.tabbed,
  org.talend.commons.ui,
  org.talend.commons,
- org.eclipse.gmf.runtime.diagram.ui.properties
+ org.eclipse.gmf.runtime.diagram.ui.properties,
+ org.talend.libraries.sdi;bundle-version="0.0.1"
 Export-Package: org.talend.core,
  org.talend.core.context,
  org.talend.core.database,
